README.md

```diff
@@ -1,2 +1,422 @@
+        self.securePersistentStorage = securePersistentStorage
+    }
+
+    // MARK: UserService
+
+    public private(set) lazy var user: User? = loadPersistedUser() {
+        didSet {
+            persistUserData()
+        }
+    }
+
+    public func login(username: String, password: String) async throws -> User {
+        let user = try await authService.login(username: username, password: password)
+        self.user = user
+        return user
+    }
+
+    // MARK: Private
+
+    /// An auth service instance that is instantiated when the `DefaultUserService` is instantiated.
+    @Instantiated
+    private let authService: AuthService
+
+    /// An instance of secure, persistent storage that is instantiated further up the SafeDI dependency tree.
+    @Received
+    private let securePersistentStorage: SecurePersistentStorage
+
+    private func loadPersistedUser() -> User? {
+        securePersistentStorage["user", ofType: User.self]
+    }
+
+    private func persistUserData() {
+        securePersistentStorage["user"] = user
+    }
+}
+```
+
+### @ExternalInstantiable
+
+Types that are declared outside of your project can be instantiated by SafeDI if there is an extension on the type decorated with the [`@ExternalInstantiable` macro](Sources/SafeDI/PropertyDecoration/ExternalInstantiable.swift). Extensions decorated with this macro define how to instantiate the extended type via a `public func instantiate(…) -> ExtendedType` method. This `instantiate(…)` method can receive dependencies instantiated or forwarded by objects further up the dependency tree by declaring these dependencies as method arguments.
+
+#### Example
+
+Here is a sample `SecurePersistentStorage` protocol whose concrete type is defined in a third-party dependency, and therefore is `@ExternalInstantiable`:
+
+```swift
+import Foundation
+import SafeDI
+import Valet // A Keychain wrapper that can be used to implement secure, persistent storage. github.com/square/valet
+
+/// A protocol defining how to interact with secure, persistent storage.
+/// It is not necessary to utilize protocols with SafeDI, but since
+/// protocol-driven development aids both testability and dependency
+/// inversion, our examples show how protocols can be used with SafeDI.
+protocol SecurePersistentStorage {
+    subscript<CodableType: Codable>(_ key: String, ofType type: CodableType.Type) -> CodableType? { get }
+    subscript<CodableType: Codable>(_ key: String) -> CodableType? { get set }
+}
+
+/// A default implementation of `SecurePersistentStorage` that can fulfill
+/// `@Instantiated` properties of type `Valet` or `SecurePersistentStorage`.
+@ExternalInstantiable(fulfillingAdditionalTypes: [SecurePersistentStorage.self])
+extension Valet: SecurePersistentStorage {
+
+    /// A public initializer defines how SafeDI can instantiate a Valet object.
+    public static func instantiate() -> Valet {
+        Valet.valet(
+            with: Identifier(nonEmpty: "SafeDIExample")!,
```

Comment:
Is it possible for Macros to deduce the identifier from the type name? Since most protocols typically have a sole default concrete implementation, adopting this approach would significantly enhance the readability of the API. 

Also, to facilitate understanding for beginners, I would recommend extending the earlier comment by simplifying the example. Specifically, suggesting a single method without parameters, such as public static func instantiate() -> Type, would make its clarity and purpose evident to the reader.

Reply:
Identifiers are a Valet-specific thing... I should probably use the `UserDefaults` example from the example project rather than inventing a newer, more complicated example here 😅. Good thinking that this is a bit more complex than it needed to be.